src/utils/protocol/downloadProtocol.js

```diff
@@ -23,27 +27,45 @@ const downloadProtocol = inEnvironment((environment) => {
     return (uri) => {
       const tempPath = (electron.app || electron.remote.app).getPath('temp');
 
-      return getProtocolNameFromUri(uri)
-        .then(protocolName => path.join(tempPath, protocolName))
-        .then(destination => writeStream(destination, request(uri)));
+      return getURL(uri)
```

Comment:
There's another case here that isn't caught; I'm not sure why. It's described by the SSL trust case in #331 but happens more generally any time a remote server can't be reached.

Test by hitting a well-formed URL at `localhost:7788` (any non-running local port) or some hostname known to not exist.

In these cases, an uncaught error shows in the console and UI spins indefinitely until reload. If we expect users to be able to import from arbitrary servers, it would be good to catch this case.

(I'm on electron/macOS.)

Reply:
Great catch!